src/stateMatrixParser.cpp

```diff
@@ -0,0 +1,443 @@
+  BestKMarkovTables<NormalizedCompressionMarkovTable> bestMkvTableCompression(kvector, alphabet_size);
+  StateMatrix stMatrix(states, alphabet_size);
+  stMatrix.set_random(rng);
+  while(! machine_filter(stMatrix, states, alphabet_size, tm_tape_iterations, amplitude_criteria,bestMkvTableCompression) ){
+        stMatrix.set_random(rng);
+  }
+  auto vector_written_string = recreate_tm(stMatrix, states, alphabet_size, tm_tape_iterations);
+  values_initial = bestMkvTableCompression.update_string(vector_written_string);
+  auto amplitude = values_initial.amplitude;
+  auto nc = values_initial.normalizedCompression;
+  bestMkvTableCompression.reset();
+
+  auto rules = uniform_int_distribution<unsigned int>(0, (alphabet_size * states)- 1);
+  unsigned int rule_index;
+  for (auto counter = 0u; counter < number_tm_it; ++counter){
+    rule_index = rules(rng);
+    auto tr_original = stMatrix.get_element(rule_index);
+    auto tr_random = set_random(rng, states, alphabet_size);
+    while(tr_original==tr_random){
+      Rng rng2{rnd_device()};
+      tr_random = set_random(rng2, states, alphabet_size);
+    }
+    stMatrix.set_rule(rule_index, tr_random);
+    vector_written_string = recreate_tm(stMatrix, states, alphabet_size, tm_tape_iterations);
+    Metrics results = bestMkvTableCompression.update_string(vector_written_string);
+    bestMkvTableCompression.reset();
+    auto amplitude2 = results.amplitude;
+    auto nc2 = results.normalizedCompression;
+    if ( (nc2 <= nc) || (amplitude2*2 < amplitude || amplitude2 < 100) ){
+      stMatrix.set_rule(rule_index, tr_original);
+    }
+    else{
+      nc = nc2;
+      amplitude = amplitude2;
+    }
+  }
+  vector_written_string = recreate_tm(stMatrix, states, alphabet_size, tm_tape_iterations);
+  values_final = bestMkvTableCompression.update_string(vector_written_string);
+  bestMkvTableCompression.reset();
+  if (print){
+  std::cout << values_initial.amplitude << "\t" << values_initial.selfCompression << "\t" << values_initial.normalizedCompression << "\t" << 
+  values_final.amplitude << "\t" << values_final.selfCompression << "\t" << values_final.normalizedCompression << std::endl; 
+  }
+  Metrics variation;
+  variation.amplitude = values_final.amplitude;
+  variation.normalizedCompression = (values_final.normalizedCompression - values_initial.normalizedCompression);
+  variation.selfCompression = (values_final.selfCompression - values_initial.selfCompression);
+  
+  return variation;
+//the algorithm:
+// 1)Create random tm;
+// 2)Randomly select rule index;
+// 3)Permute randomly rule;
+// 4)Compute NC and Amplitude;
+// 5) if it does not improve NC higher and Amplitude higher than 100 chars, permute rule;
+// 6)it improves, yes change rule repeat 2-4 until numb_tm_it;
+// 7) print tape and state matrix;
+}
+
+
+void test(){
+  //evolve_multiple_tms_graph(3200,8000,10,2);
+  real_nc_evolution(200000,5000, 10, 2,false);
+  exit(55);
+  evolve_multiple_tms2(2000,5000,10,2);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statements following return, break, continue, goto or throw will never be executed.](https://app.codacy.com/app/jorgeMFS/TMCompression/pullRequest?prid=4494377)